vertx-grpc-context-storage/src/test/java/io/vertx/grpc/context/storage/ContextStorageTest.java

```diff
@@ -0,0 +1,139 @@
+    CallOptions.Key<String> traceOptionsKey = CallOptions.Key.create("trace");
+    Key<String> traceMetadataKey = Key.of("trace", ASCII_STRING_MARSHALLER);
+    Context.Key<String> traceContextKey = Context.key("trace");
+
+    GreeterGrpc.GreeterImplBase impl = new GreeterGrpc.GreeterImplBase() {
+      @Override
+      public void sayHello(HelloRequest request, StreamObserver<HelloReply> responseObserver) {
+        vertx.<String>executeBlocking(prom -> {
+          prom.complete("Hello " + request.getName() + ", trace: " + traceContextKey.get());
+        }).onSuccess(greeting -> {
+          responseObserver.onNext(HelloReply.newBuilder().setMessage(greeting).build());
+          responseObserver.onCompleted();
+        }).onFailure(should::fail);
+      }
+    };
+
+    ServerServiceDefinition def = ServerInterceptors.intercept(impl, new ServerInterceptor() {
+      @Override
+      public <ReqT, RespT> ServerCall.Listener<ReqT> interceptCall(ServerCall<ReqT, RespT> call, Metadata headers, ServerCallHandler<ReqT, RespT> next) {
+        String traceId = headers.get(traceMetadataKey);
+        should.assertNotNull(traceId);
+        Context context = Context.current().withValue(traceContextKey, traceId);
+        Context previous = context.attach();
+        return new ForwardingServerCallListener.SimpleForwardingServerCallListener<ReqT>(next.startCall(call, headers)) {
+          @Override
+          public void onComplete() {
+            context.detach(previous);
+            super.onComplete();
+          }
+        };
+      }
+    });
+
+    GrpcServer server = GrpcServer.server(vertx);
+    GrpcServiceBridge serverStub = GrpcServiceBridge.bridge(def);
+    serverStub.bind(server);
+
+    Async servertStart = should.async();
+    vertx.createHttpServer()
+      .requestHandler(server)
+      .listen(0, should.asyncAssertSuccess(httpServer -> {
+        this.httpServer = httpServer;
+        servertStart.complete();
+      }));
+    servertStart.await();
+
+    channel = ManagedChannelBuilder.forAddress("localhost", httpServer.actualPort())
+      .usePlaintext()
+      .build();
+
+    GreeterGrpc.GreeterBlockingStub stub = GreeterGrpc.newBlockingStub(ClientInterceptors.intercept(channel, new ClientInterceptor() {
+        @Override
+        public <ReqT, RespT> ClientCall<ReqT, RespT> interceptCall(MethodDescriptor<ReqT, RespT> method, CallOptions callOptions, Channel next) {
+          return new ForwardingClientCall.SimpleForwardingClientCall<ReqT, RespT>(next.newCall(method, callOptions)) {
+            @Override
+            public void start(Listener<RespT> responseListener, Metadata headers) {
+              Metadata traceHeaders = new Metadata();
+              traceHeaders.put(traceMetadataKey, callOptions.getOption(traceOptionsKey));
+              headers.merge(traceHeaders);
+              super.start(responseListener, headers);
+            }
+          };
+        }
+      }))
+      .withCompression("identity");
```

Comment:
Actually, it's not possible because we need to extract the trace key from call options in `traceHeaders.put(traceMetadataKey, callOptions.getOption(traceOptionsKey));`